mne/io/curry/curry.py

```diff
@@ -0,0 +1,313 @@
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to a curry file with extensions .dat, .dap, .rs3, .cdt, cdt.dpa,
+        .cdt.cef or .cef.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    %(verbose)s
+
+    Returns
+    -------
+    raw : instance of RawCurry
+        A Raw object containing Curry data.
+
+    """
+    return RawCurry(input_fname, preload, verbose)
+
+
+class RawCurry(BaseRaw):
+    """Raw object from Curry file.
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to a curry file with extensions .dat, .dap, .rs3, .cdt, cdt.dpa,
+        .cdt.cef or .cef.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    %(verbose)s
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+
+    """
+
+    def __init__(self, input_fname, preload=False, verbose=None):
+
+        # we don't use os.path.splitext to also handle extensions like .cdt.dpa
+        fname_base, ext = input_fname.split(".", maxsplit=1)
+        curry_vers = _get_curry_version(ext)
+        _check_missing_files(fname_base, curry_vers)
+        data_fname = os.path.abspath(fname_base +
+                                     DATA_FILE_EXTENSION[curry_vers])
+
+        info, n_samples, is_ascii = _read_curry_info(fname_base, curry_vers)
+
+        last_samps = [n_samples - 1]
+        self._is_ascii = is_ascii
+
+        super(RawCurry, self).__init__(
+            info, preload, filenames=[data_fname], last_samps=last_samps,
+            orig_format='int', verbose=verbose)
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a chunk of raw data."""
+        if self._is_ascii:
+            ch_idx = range(0, len(self.ch_names))
+            if version.parse(np.version.version) >= version.parse("1.16.0"):
+                block = np.loadtxt(self.filenames[0],
+                                   skiprows=start,
+                                   usecols=ch_idx[idx],
+                                   max_rows=stop - start).T
+            else:
+                block = np.loadtxt(self.filenames[0],
+                                   skiprows=start,
+                                   usecols=ch_idx[idx]).T
+                block = block[:, :stop - start]
```

Comment:
Should it give a warning here, that more data is loaded as expected?
e.g. "Data reading might take longer for ASCII files. Update numpy to version 1.16.0 or greater for more efficient data reading."

Reply:
Sure